src/types/Cache.ts

```diff
@@ -0,0 +1,15 @@
+import { SecondLevelDomain } from "./SecondLevelDomain";
+import { TopLevelDomain } from "./TopLevelDomain";
+
+export interface ICache {
+  getTld(tldHash: string): Promise<TopLevelDomain> | TopLevelDomain | undefined;
+  getTlds(): Promise<TopLevelDomain[]> | TopLevelDomain[] | undefined;
```

Comment:
`Promise<TopLevelDomain[] | undefined>` instead of `Promise<TopLevelDomain[]> | TopLevelDomain[] | undefined`.